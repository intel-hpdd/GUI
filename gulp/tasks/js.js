//
// INTEL CONFIDENTIAL
//
// Copyright 2013-2016 Intel Corporation All Rights Reserved.
//
// The source code contained or described herein and all documents related
// to the source code ("Material") are owned by Intel Corporation or its
// suppliers or licensors. Title to the Material remains with Intel Corporation
// or its suppliers and licensors. The Material contains trade secrets and
// proprietary and confidential information of Intel or its suppliers and
// licensors. The Material is protected by worldwide copyright and trade secret
// laws and treaty provisions. No part of the Material may be used, copied,
// reproduced, modified, published, uploaded, posted, transmitted, distributed,
// or disclosed in any way without Intel's prior express written permission.
//
// No license under any patent, copyright, trade secret or other intellectual
// property right is granted to or conferred upon you by disclosure or delivery
// of the Materials, either expressly, by implication, inducement, estoppel or
// otherwise. Any license under such intellectual property rights must be
// express and approved by Intel in writing.

'use strict';

var paths = require('../paths.json');
var gulp = require('gulp');
var babel = require('gulp-babel');
var sourcemaps = require('gulp-sourcemaps');
var destDir = require('../dest-dir');
var relativeSourcemapsSource = require('gulp-relative-sourcemaps-source');

function jsDeps (fn) {
  return gulp.src(paths.js.deps, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.jsDepsDev = function jsDepsDev () {
  return jsDeps(jsDepsDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsDepsProd = function jsDepsProd () {
  return jsDeps(jsDepsProd)
  .pipe(gulp.dest('./dest'));
};

function socketWorker (fn) {
  return gulp.src(paths.js.socketWorker, {
    since: gulp.lastRun(fn)
  })
  .pipe(sourcemaps.init({ loadMaps: true }))
  .pipe(sourcemaps.write('.'));
}

exports.socketWorkerDev = function socketWorkerDev () {
  return socketWorker(socketWorkerDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.socketWorkerProd = function socketWorkerProd () {
  return socketWorker(socketWorkerProd)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.dest('./dist'));
};

function systemConfig (fn) {
  return gulp.src(paths.js.config, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.systemConfigDev = function systemConfigDev () {
  return systemConfig(systemConfigDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

var babelDev = babel.bind(null, {
  presets: [],
  babelrc: false,
  plugins: [
    'check-es2015-constants',
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    'transform-async-to-generator',
    'transform-es2015-classes',
    'transform-es2015-modules-systemjs'
  ]
});

var babelProd = babel.bind(null, {
  presets: ['es2015'],
  babelrc: false,
  plugins: [
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    [
      'angularjs-annotate',
      {
        explicitOnly: true
      }
    ],
    'transform-es2015-modules-systemjs'
  ]
});

function jsSource (fn) {
  return gulp.src(paths.js.source, {
    since: gulp.lastRun(fn),
    base: '.'
  })
  .pipe(sourcemaps.init());
}

exports.jsSourceDev = function jsSourceDev () {
  return jsSource(jsSourceDev)
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '/static/chroma_ui' }))
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsSourceProd = function jsSourceProd () {
  return jsSource(jsSourceProd)
  .pipe(babelProd())
  .pipe(relativeSourcemapsSource({dest: 'dest'}))
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTest = function jsTest () {
  return gulp.src(paths.js.tests, {
    since: gulp.lastRun(jsTest),
    base: '.'
  })
  .pipe(sourcemaps.init())
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTestFixtures = function jsTestFixtures () {
  return gulp.src(paths.js.testFixtures, {
    since: gulp.lastRun(jsTestFixtures),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};

exports.jsTestDeps = function jsTestDeps () {
  return gulp.src(paths.js.testDeps, {
    since: gulp.lastRun(jsTestDeps),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};
