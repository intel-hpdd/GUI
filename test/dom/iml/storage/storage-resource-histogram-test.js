// @flow

import StorageResourceHistogram from "../../../../source/iml/storage/storage-resource-histogram.js";
import { renderToSnapshot } from "../../../test-utils.js";

it("should render as expected", () => {
  const data = {
    type: "histogram",
    title: "my first histogram",
    series: [
      {
        label: "reads",
        name: "reads",
        type: "histogram",
        data: {
          bin_labels: [">1", ">2", ">3"],
          values: [2, 3, 4]
        },
        unit_name: "reads"
      }
    ]
  };

  expect(renderToSnapshot(<StorageResourceHistogram chart={data} />)).toMatchSnapshot();
});

it("should render as expected with two series", () => {
  const data = {
    type: "histogram",
    title: "my second histogram",
    series: [
      {
        label: "reads",
        name: "reads",
        type: "histogram",
        data: {
          bin_labels: [">1", ">2", ">3"],
          values: [2, 3, 4]
        },
        unit_name: "reads"
      },
      {
        label: "writes",
        name: "writes",
        type: "histogram",
        data: {
          bin_labels: [">1", ">2", ">3"],
          values: [2.5, 3.5, 4.5]
        },
        unit_name: "writes"
      }
    ]
  };

  expect(renderToSnapshot(<StorageResourceHistogram chart={data} />)).toMatchSnapshot();
});
