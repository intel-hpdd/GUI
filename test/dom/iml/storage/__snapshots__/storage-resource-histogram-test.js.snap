// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as expected 1`] = `
<div>
  <div
    class="storage-detail-chart"
    style="height: 500px;"
  >
    <svg
      class="charting"
      style="width: 100%; height: 100%;"
    >
      <g
        class="charting-group"
        transform="translate(50,50)"
      >
        <g
          class="axis x-axis"
          transform="translate(0,-100)"
        >
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,0)"
          >
            <line
              y2="6"
              x2="0"
            />
            <text
              dy=".71em"
              style="text-anchor: middle;"
              y="9"
              x="0"
            >
              >1
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(-40,0)"
          >
            <line
              y2="6"
              x2="0"
            />
            <text
              dy=".71em"
              style="text-anchor: middle;"
              y="9"
              x="0"
            >
              >2
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(-80,0)"
          >
            <line
              y2="6"
              x2="0"
            />
            <text
              dy=".71em"
              style="text-anchor: middle;"
              y="9"
              x="0"
            >
              >3
            </text>
          </g>
          <path
            class="domain"
            d="M-80,6V0H0V6"
          />
        </g>
        <g
          class="axis y-axis"
        >
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-100)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              0
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-90.38461538461539)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              10
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-80.76923076923077)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              20
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-71.15384615384616)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              30
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-61.53846153846154)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              40
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-51.92307692307692)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              50
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-42.307692307692314)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              60
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-32.692307692307686)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              70
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-23.076923076923073)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              80
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-13.461538461538458)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              90
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-3.8461538461538436)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              100
            </text>
          </g>
          <path
            class="domain"
            d="M-6,-100H0V0H-6"
          />
        </g>
        <defs
          class="area-clip1"
        >
          <clippath
            id="area-clip1"
          >
            <rect
              width="-80"
              height="-100"
            />
          </clippath>
        </defs>
        <g
          clip-path="url(about:blank#area-clip1)"
          class="clipPath1"
        >
          <path
            class="area area1"
            stroke="#1f77b4"
            fill="#1f77b4"
            fill-opacity="0.2"
            d="M0,-98.07692307692307Q-32,-97.3076923076923,-40,-97.11538461538461Q-48,-96.92307692307692,-80,-96.15384615384616L-80,-100Q-48,-100,-40,-100Q-32,-100,0,-100Z"
          />
        </g>
        <g>
          
        </g>
      </g>
      <g
        class="legend chart-legend"
        transform="translate(50,0)"
      >
        <g
          class="legend-group"
          transform="translate(0,0)"
        >
          <circle
            class="legend-circle"
            fill="#1f77b4"
            stroke="#1f77b4"
            fill-opacity="0"
            r="5"
          />
          <text
            class="legend-label"
            dx="10"
            dy="4"
          >
            reads
          </text>
        </g>
      </g>
    </svg>
  </div>
</div>
`;

exports[`should render as expected with two series 1`] = `
<div>
  <div
    class="storage-detail-chart"
    style="height: 500px;"
  >
    <svg
      class="charting"
      style="width: 100%; height: 100%;"
    >
      <g
        class="charting-group"
        transform="translate(50,50)"
      >
        <g
          class="axis x-axis"
          transform="translate(0,-100)"
        >
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,0)"
          >
            <line
              y2="6"
              x2="0"
            />
            <text
              dy=".71em"
              style="text-anchor: middle;"
              y="9"
              x="0"
            >
              >1
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(-40,0)"
          >
            <line
              y2="6"
              x2="0"
            />
            <text
              dy=".71em"
              style="text-anchor: middle;"
              y="9"
              x="0"
            >
              >2
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(-80,0)"
          >
            <line
              y2="6"
              x2="0"
            />
            <text
              dy=".71em"
              style="text-anchor: middle;"
              y="9"
              x="0"
            >
              >3
            </text>
          </g>
          <path
            class="domain"
            d="M-80,6V0H0V6"
          />
        </g>
        <g
          class="axis y-axis"
        >
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-100)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              0
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-90.43062200956938)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              10
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-80.86124401913875)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              20
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-71.29186602870814)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              30
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-61.72248803827751)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              40
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-52.15311004784689)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              50
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-42.58373205741627)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              60
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-33.01435406698564)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              70
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-23.444976076555022)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              80
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-13.875598086124397)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              90
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,-4.30622009569378)"
          >
            <line
              x2="-6"
              y2="0"
            />
            <text
              dy=".32em"
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              100
            </text>
          </g>
          <path
            class="domain"
            d="M-6,-100H0V0H-6"
          />
        </g>
        <defs
          class="area-clip2"
        >
          <clippath
            id="area-clip2"
          >
            <rect
              width="-80"
              height="-100"
            />
          </clippath>
        </defs>
        <g
          clip-path="url(about:blank#area-clip2)"
          class="clipPath2"
        >
          <path
            class="area area2"
            stroke="#1f77b4"
            fill="#1f77b4"
            fill-opacity="0.2"
            d="M0,-98.08612440191388Q-32,-97.32057416267942,-40,-97.1291866028708Q-48,-96.93779904306218,-80,-96.17224880382776L-80,-100Q-48,-100,-40,-100Q-32,-100,0,-100Z"
          />
        </g>
        <defs
          class="area-clip3"
        >
          <clippath
            id="area-clip3"
          >
            <rect
              width="-80"
              height="-100"
            />
          </clippath>
        </defs>
        <g
          clip-path="url(about:blank#area-clip3)"
          class="clipPath3"
        >
          <path
            class="area area3"
            stroke="#ff7f0e"
            fill="#ff7f0e"
            fill-opacity="0.2"
            d="M0,-97.60765550239235Q-32,-96.8421052631579,-40,-96.65071770334929Q-48,-96.45933014354067,-80,-95.69377990430623L-80,-100Q-48,-100,-40,-100Q-32,-100,0,-100Z"
          />
        </g>
        <g>
          
        </g>
      </g>
      <g
        class="legend chart-legend"
        transform="translate(50,0)"
      >
        <g
          class="legend-group"
          transform="translate(0,0)"
        >
          <circle
            class="legend-circle"
            fill="#1f77b4"
            stroke="#1f77b4"
            fill-opacity="0"
            r="5"
          />
          <text
            class="legend-label"
            dx="10"
            dy="4"
          >
            reads
          </text>
        </g>
        <g
          class="legend-group"
          transform="translate(0,0)"
        >
          <circle
            class="legend-circle"
            fill="#ff7f0e"
            stroke="#ff7f0e"
            fill-opacity="0"
            r="5"
          />
          <text
            class="legend-label"
            dx="10"
            dy="4"
          >
            writes
          </text>
        </g>
      </g>
    </svg>
  </div>
</div>
`;
